File: [Manya]/Leetcode/Sorting/945.java
package Sorting;

class Solution {
    public int minIncrementForUnique(int[] nums) {
        int maxElement = Integer.MIN_VALUE;
        for(int i = 0; i < nums.length; i++){
            maxElement = Math.max(maxElement, nums[i]);
        }
        int extra = 0;
        int move = 0;

        int[] arr = new int[nums.length + maxElement];

        for(int i : nums){
            arr[i]++;
        }

        for(int i = 0; i < arr.length-1; i++){
            if(arr[i] > 1){
                extra = arr[i] - 1;
                arr[i+1] += extra;
                move += extra;                
            }
        }
        return move;
    }
}

File: Aditya-Pratap/Sorting/945.py
class Solution:
    def minIncrementForUnique(self, nums: List[int]) -> int:
        n = len(nums)
        m = max(nums) if nums else 0
        arr = [0] * (n + m + 1)
        
        for num in nums:
            arr[num] += 1
        
        score = 0
        
        for i in range(len(arr) - 1):
            if arr[i] > 1:
                score += arr[i] - 1
                arr[i + 1] += arr[i] - 1
        
        return score